manifests/globals.pp

```diff
@@ -1,5 +1,81 @@
-# Class for setting cross-class global overrides. See README.md for more
-# details.
+# @summary Class for setting cross-class global overrides.
+#
+# @note
+#   Most server-specific defaults should be overridden in the postgresql::server class. 
+#   This class should be used only if you are using a non-standard OS, or if you are changing elements that can only be changed here, such as version or manage_package_repo.
+#
+#
+# @param client_package_name  Overrides the default PostgreSQL client package name.    
+# @param server_package_name Overrides the default PostgreSQL server package name.
+# @param contrib_package_name Overrides the default PostgreSQL contrib package name.
+# @param devel_package_name Overrides the default PostgreSQL devel package name.
+# @param java_package_name Overrides the default PostgreSQL java package name.
+# @param docs_package_name Overrides the default PostgreSQL docs package name.
+# @param perl_package_name Overrides the default PostgreSQL Perl package name.
+# @param plperl_package_name Overrides the default PostgreSQL PL/Perl package name.
+# @param plpython_package_name Overrides the default PostgreSQL PL/Python package name.
+# @param python_package_name Overrides the default PostgreSQL Python package name.
+# @param postgis_package_name Overrides the default PostgreSQL PostGIS package name.     
+#
+# @param service_name Overrides the default PostgreSQL service name.
+# @param service_provider Overrides the default PostgreSQL service provider.
+# @param service_status Overrides the default status check command for your PostgreSQL service.
+# @param default_database Specifies the name of the default database to connect with.
+#
+# @param validcon_script_path     
+#
+# @param initdb_path Path to the initdb command.
+# @param createdb_path Deprecated. Path to the createdb command.
+# @param psql_path Sets the path to the psql command.
+# @param pg_hba_conf_path Specifies the path to your pg_hba.conf file.
+# @param pg_ident_conf_path Specifies the path to your pg_ident.conf file.
+# @param postgresql_conf_path Sets the path to your postgresql.conf file.
+# @param recovery_conf_path Path to your recovery.conf file.
+# @param default_connect_settings
+#
+# @param pg_hba_conf_defaults Disables the defaults supplied with the module for pg_hba.conf if set to false. This is useful if you want to override the defaults. Be sure that your changes align with the rest of the module, as some access is required to perform some operations, such as basic psql operations.    
```

Comment:
A single line should be no more than 140 characters. Ensure puppet strings is ran with `--debug` flag.